### PR TITLE
Raise an error for unsupported JULIA_CPU_TARGET

### PR DESCRIPTION
**Pull request by nalimilan**
Extend the existing check for `MARCH`.

### CODE REVIEW

File: Make.inc [hunk 1]
```diff
@@ -655,6 +655,11 @@ $(error Pre-SSE2 CPU targets not supported. To create a generic 32-bit x86 binar
 pass 'MARCH=pentium4'.)
 endif
 
+ifneq ($(findstring $(JULIA_CPU_TARGET),generic i386 i486 i586 i686 pentium pentium2 pentium3),)
```
Comment (staticfloat) [inline Make.inc]: We don't want to reject `generic`; that's what we actually use on `aarch64`.

**Review by staticfloat — CHANGES_REQUESTED** [review]

File: Make.inc (hunk 1 above)
Comment (nalimilan) [reply Make.inc]: Good catch. Ideally we should check whether the arch is x86 and reject `generic` only in that case, but given that it's only for 0.6 and we have a more reliable solution on master, let's go with the simpler fix.

Comment (nalimilan) [reply Make.inc]: Thanks, but at this point is it really worth taking the risk of breaking things on 0.6? Maybe better go with the more limited patch.

**Comment by nalimilan** [comment]
@ararslan If you feel like including this in 0.6.4, please do so, otherwise just close this PR as it doesn't matter a lot at this point.

**Comment by ararslan** [comment]
Thanks for this. I'm just going to be careful for the 0.6.4 builds though, as I don't want to change the state of the release or backports branches unless absolutely necessary at this point.

**Pull request closed without merging**